test/provision/k8s_install.sh

```diff
@@ -26,7 +26,10 @@ export KUBEADM_OPTIONS=""
 export K8S_FULL_VERSION=""
 export DNS_DEPLOYMENT="${PROVISIONSRC}/manifest/dns_deployment.yaml"
 export KUBEDNS_DEPLOYMENT="${PROVISIONSRC}/manifest/kubedns_deployment.yaml"
-export COREDNS_DEPLOYMENT="${PROVISIONSRC}/manifest/coredns_deployment.yaml"
+export COREDNS_DEPLOYMENT="${PROVISIONSRC}/manifest/${K8S_VERSION}/coredns_deployment.yaml"
```

Comment:
Why we don't use the normal coredns deployment file? What changes? Kubeadm always use the same one no?

Reply:
@eloycoto coredns is different between k8s versions.

Reply:
Hi, If you see kubeadm, as the file that we have now, is the same for all versions: 

https://raw.githubusercontent.com/kubernetes/kubernetes/master/cmd/kubeadm/app/phases/addons/dns/manifests.go

And we have used the same strategy for a few months now, I'm not sure what is the value added if we have to maintain different versions of corednds across all kubernetes versions. If Kubeadm, that is deployed by kubernetes maintainers, follow that behaviour I do not see anything wrong with that. 

IMHO.

Reply:
@eloycoto that file might look the same but that file is a template, in reality the coredns version is not same.

https://github.com/kubernetes/kubernetes/blob/release-1.11/cmd/kubeadm/app/constants/constants.go#L286-L287

https://github.com/kubernetes/kubernetes/blob/release-1.12/cmd/kubeadm/app/constants/constants.go#L304-L305

Reply:
With kubeadm 1.12 if you install kubernetes 1.10 will install the latest version of coredns, and that for us is not bad at all. We don't want to debug core-dns issues. 

If we try to keep the same version for each kubernetes we will need to have different manifest, maintain it, test that may fail because know issues of core-dns. I still don't know what is the benefit. 

core-dns for us a is a component, I would not spend time on that, keep the last one, as kubeadm is doing.

Reply:
@eloycoto that's not how users will deploy kubernetes, is it? If they have kubeadm 1.11 they will deploy the coredns version 1.11 as well. Do we want to test what users deploy or what make our lives easier?

Reply:
I mean, It is not make our life easier, is making the Cilium reliable and no debug CoreDNS issues. We need to make sure that connectivity works, no that Coredns works, for me is a different behaviour, we need to focus in what Cilium do, making sure that connectivity and policy enforcement works correctly and try to avoid flakes related to CoreDNS versions.

Reply:
I understand both arguments. I don't think there is a right or wrong here. There are risks with both older versions and latest. Latest can also introduce bug which we then account to Cilium changes.

I would slightly prefer extended test coverage by not just using a single coredns version.